urllib3/contrib/pyopenssl.py

```diff
@@ -223,10 +231,12 @@ def get_subj_alt_name(peer_cert):
     # Sadly the DNS names need to be idna encoded and then, on Python 3, UTF-8
     # decoded. This is pretty frustrating, but that's what the standard library
     # does with certificates, and so we need to attempt to do the same.
-    names = [
-        ('DNS', _dnsname_to_stdlib(name))
-        for name in ext.get_values_for_type(x509.DNSName)
-    ]
+    # We also want to skip over names which cannot be idna encoded.
+    names = []
+    for name in ext.get_values_for_type(x509.DNSName):
```

Comment:
We could instead do

```py
names = [
    ('DNS', name) for name in map(_dnsname_to_stdlib, ext.get_values_for_type(x509.DNSName))
    if name is not None
]
```

Reply:
I'll change to that! I never think to use `map`. :)